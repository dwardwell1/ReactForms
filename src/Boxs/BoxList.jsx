import {useState} from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import {v4 as uuid} from "uuid"; 

const BoxList = () => {
    const INITIAL_STATE = [
        {id:uuid(), color:"black", height:20, width:20},
        {id:uuid(), color:"black", height:30, width:30}]
    const [boxes, setBoxes] = useState(INITIAL_STATE);
    const addBox = (newBox) => { 
        setBoxes(boxes => [...boxes, {...newBox, id:uuid()}]);
    }
    const removeBox = (boxId) => {
        setBoxes(boxes => boxes.filter(b => b.id !== boxId));
    }
    
    return (    
        <div>
            <h3>Check Out These Boxes</h3>
            <NewBoxForm addBox={addBox} />
            <div>
                {boxes.map(({id,color, height, width}) => (<Box id={id} color={color} width={width} height={height}  removeBox={removeBox}/>))}
            </div>
        </div>
)}

export default BoxList;
