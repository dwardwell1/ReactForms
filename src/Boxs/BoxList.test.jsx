import {render, fireEvent, getByLabelText, findAllByText, getAllByText} from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function(){
    render(<BoxList />);
});

it("matches snapshot", function(){
    const {asFragment} = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add item to shopping list", function(){
    const {queryByText, getByLabelText} = render(<BoxList />);
    const Color = getByLabelText("Choose Color");
    const Width = getByLabelText("Width:");
    const Height = getByLabelText("Height:");
    const btn = queryByText("Add")
    // const buttons = getAllByText("x")
    // expect(buttons.length).toEqual(2)
    fireEvent.change(Color, {target: {value: "#9d4848"}});
    fireEvent.change(Width, {target: {value: "200"}});
    fireEvent.change(Height, {target: {value: "130"}});
    fireEvent.click(btn);
    expect(queryByText("#9d4848")).toBeDefined();
    // expect(buttons.length).toEqual(3)

});