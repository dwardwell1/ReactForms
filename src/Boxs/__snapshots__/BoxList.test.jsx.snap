// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Check Out These Boxes
    </h3>
    <form>
      <label
        for="color"
      >
        Choose Color
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value=""
      />
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        id="height"
        max="200"
        min="10"
        name="height"
        type="range"
        value=""
      />
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        id="width"
        max="200"
        min="10"
        name="width"
        type="range"
        value=""
      />
      <input
        type="submit"
        value="Add"
      />
    </form>
    <div>
      <div
        class="BoxHolder"
      >
        <div
          class="box"
          style="height: 20px; width: 20px; background-color: black; display: inline-block;"
        />
        <button>
          x
        </button>
      </div>
      <div
        class="BoxHolder"
      >
        <div
          class="box"
          style="height: 30px; width: 30px; background-color: black; display: inline-block;"
        />
        <button>
          x
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
